src/cpp/server/server_cc.cc

```diff
@@ -1201,6 +1119,30 @@ int Server::AddListeningPort(const grpc::string& addr,
   return port;
 }
 
+void Server::Ref() {
+  shutdown_refs_outstanding_.fetch_add(1, std::memory_order_relaxed);
+}
+
+void Server::UnrefWithPossibleNotify() {
+  if (shutdown_refs_outstanding_.fetch_add(-1, std::memory_order_acq_rel) ==
```

Comment:
nit: I would use `fetch_sub(1, ...)` here and below.

Also it would be worth adding GPR_UNLIKELY here.

Reply:
Done.